packages/client/src/components/SendTokens/steps/SendTokenForm.js

```diff
@@ -1,23 +1,28 @@
-import { useContext } from "react";
-import { ASSET_TYPES } from "constants/enums";
-import { SendTokensContext } from "../sendTokensContext";
-import SendModalHeader from "../components/SendModalHeader";
-import AmountInput from "../components/AmountInput";
-import AddressDropdown from "../components/AddressDropdown";
-import AssetSelector from "../components/AssetSelector";
-import ButtonGroup from "../components/ButtonGroup";
+import { useContext } from 'react';
+import { Web3Context } from 'contexts/Web3';
+import AddressDropdown from '../components/AddressDropdown';
+import AmountInput from '../components/AmountInput';
+import AssetSelector from '../components/AssetSelector';
+import { ASSET_TYPES } from 'constants/enums';
+import { SendTokensContext } from '../sendTokensContext';
 
 const SendTokenForm = () => {
   const [sendModalState] = useContext(SendTokensContext);
-  const { assetType } = sendModalState;
+  const { assetType, address } = sendModalState;
```

Comment:
In SendTokenForm I am fetching the balances now from the Web3 context
But in deposit form the balances of the user's vault should be fetched each time the deposit token is opened, to make sure the balances are correct (it can change every second, and it is not controlled by Vessel)

Reply:
What I mean is more about where to put the data after fetching (either from web3 context or from api call for user's vaults), right now we fetch data on `SendTokenForm` and `DepositTokenForm` then pass them down through the property chain via property `coinBalances`, what I'm suggesting is putting the coinBalances data into the context, and have cointypedropdown reading from the context. Let me know if it makes more sense.

Reply:
Ohhh right, makes sense! I will put it inside the context and fetch it instead of passing through property. Thank you!